### ISSUE TITLE
[A11y] Date under "Sync Now" should be read in natural format in Voice Over

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/firefox-ios#35484]
The formatter is currently using a fixed format instead of the regional format set by the user in settings:  

```
formatter.dateFormat = "yyyy-MM-dd HH:mm:ss"
```

We should use the users preferences here instead and display the correct format. This should resolve the date being read incorrectly as well.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-16736)
